### ISSUE TITLE
tntbuild 2.0.7 Error on OSX Yosemite 10.10.2

### ISSUE THREAD

**Issue by mstewart20** [trueinteractions/tint2#92]
I am following the instructions in the Getting Started guide for building my first tint app and ran into an issue while trying to build and package the application using tntbuild. It looks like it's an error inside of the tntbuild script itself, so not sure how to proceed. I've set up the directory and created the files that are specified in https://www.trueinteractions.com/tint2/docs/tutorial-gettingstarted.html

### Directory Structure:
```
mstewart:guru-tint mstewart$ ls -al
total 24
drwxr-xr-x    5 mstewart  staff   170B Jan 30 09:37 ./
drwxr-xr-x  119 mstewart  staff   4.0K Jan 30 09:28 ../
-rw-r--r--    1 mstewart  staff   588B Jan 30 09:30 index.html
-rw-r--r--    1 mstewart  staff   821B Jan 30 09:30 main.js
-rw-r--r--    1 mstewart  staff   310B Jan 30 09:29 package.json
```

### tntbuild Error:
```
mstewart:guru-tint mstewart$ tntbuild package.json

/usr/local/bin/tntbuild:40
  b.data.icon.osx[0]=$tint.absolute(b.data.icon.osx[0],sourceDirectory);
             ^
TypeError: Cannot read property 'osx' of undefined
    at Object.$tint.loadbuilder (/usr/local/bin/tntbuild:40:14)
    at Object.<anonymous> (/usr/local/bin/tntbuild:2791:19)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:942:3
```

**Comment by trevorlinton** [comment]
@mstewart20 Hi Mitchell, the error is due to not having icon resources defined in the package.json format; the example in getting started doesn't have them but the example in packaged applications does. 

This is an over-sight; you can fix it by adding an icon to your package.json file as described here: 

https://www.trueinteractions.com/tint2/docs/tutorial-packaging.html

(Or just add these lines, and the corresponding PNG images for icons to your package.json) 

```
  "icon": {
    "osx":["./tintruntime.png"],
    "windows":["./tintruntime.png"]
  },
```

For the long term - I'll leave this open and we'll fix the tntbuild script to not require an icon when building an application (as it does at the moment).

**Comment by mstewart20** [comment]
Thanks @trevorlinton, that worked! I'll dig deeper into the documentation next time, thanks for the pointers. 

You could always spit out an error message as well from tntbuild for any missing requirements, that would've been self explanatory as well.

**Comment by trevorlinton** [comment]
@mstewart20 Valid point, we're making some larger updates to it for the next V2.0.9 release next week; i'll put this into the queue for that release target.

**Comment by trevorlinton** [comment]
@mstewart20  We've fixed the tntbuild script to error out if an icon file is not specified (and it now properly checks dependencies before starting the build).  This is fixed in https://github.com/trueinteractions/tint2/commit/562c3669d547048ee7e76deb9c56f5038af8bab3 and will be released with V2.0.9 on Mon, Feb 9 2015.
